src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java

```diff
@@ -783,6 +783,7 @@ private long approximateLastCycle2(int lastCycle) throws StreamCorruptedExceptio
             if (this.store != wireStore) {
                 releaseStore();
                 this.store = wireStore;
+                this.cycle = lastCycle;
```

Comment:
We used to update the `store` field without making a corresponding change to the `cycle` field.

Reply:
this caused issues for the logic above that says:

` wireStore = (cycle == lastCycle) ? this.store : queue.storeForCycle(lastCycle, queue.epoch(), false, this.store);`

on the subsequent recursion.